src/components/form/form.types.ts

```diff
@@ -33,3 +35,67 @@ export interface FormError {
      */
     message: string;
 }
+
+export interface FormComponent<T = any> {
+    /**
+     * The value of the current property
+     */
+    value: T;
+
+    /**
+     * Whether or not the current property is required
+     */
+    required?: boolean;
+
+    /**
+     * Whether or not the current property is readonly
+     */
+    readonly?: boolean;
+
+    /**
+     * Whether or not the current property is disabled
+     */
+    disabled?: boolean;
+
+    /**
+     * The name of the current property
+     */
+    name?: string;
+
+    /**
+     * The title of the current property
+     */
+    title?: string;
+
+    /**
+     * The helper text for the current property
+     */
+    helperText?: string;
+
+    formInfo?: {
+        /**
+         * The schema of the current property
+         */
+        schema?: object;
+
+        /**
+         * The schema of the whole form
+         */
+        rootSchema?: object;
+
+        /**
+         * A tree of errors for this property and its children
+         */
+        errorSchema?: object;
+
+        /**
+         * The value of the whole form
+         */
+        rootValue?: any;
+    };
```

Comment:
It might be nice to have an interface for this as well so you could refer to it in your own code when implementing `FormComponent` :thinking:

Reply:
agreed